service/controller/resource/scrapeconfigs/resource.go

```diff
@@ -83,14 +89,33 @@ func toSecret(v interface{}, provider string) (*corev1.Secret, error) {
 	return scrapeConfigsSecret, nil
 }
 
-func renderTemplate(clusterID string, provider string) (string, error) {
-	var templateData = TemplateData{
+func getTemplateData(cluster metav1.Object, provider string, clients k8sclient.Interface) (*TemplateData, error) {
+	c, ok := cluster.(*v1alpha2.Cluster)
+	if !ok {
+		return nil, microerror.Mask(fmt.Errorf("invalid v1alpha2.Cluster %T", cluster))
+	}
+
+	infra, err := clients.G8sClient().InfrastructureV1alpha2().AWSClusters(c.Spec.InfrastructureRef.Namespace).Get(c.Spec.InfrastructureRef.Name, metav1.GetOptions{})
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	clusterID := key.ClusterID(cluster)
+	domain := infra.Spec.Cluster.DNS.Domain
```

Comment:
Can we not use the base domain for this?

Reply:
For this I've followed how the etcd url is built in was-operator: https://github.com/giantswarm/aws-operator/blob/e1111eef8fe7a32f9eeb2f23e130a308be2cd1c5/service/controller/key/cluster.go#L102

Where information are taken from the infra CR.